src/services/resource.js

```diff
@@ -210,6 +358,364 @@ module.exports = class resourceHelper {
+				order: queryParams.sort_order?.toUpperCase() === common.SORT_ASC ? common.SORT_ASC : common.SORT_DESC,
+			}
+			let finalResourceIds = []
+			let resourceIdsToBeRemoved = []
+			let resourcesIdsNotes = {}
+			let inProgressResources = []
+			let uniqueOrganizationIds = []
+
+			// check review resources and find all resources and org for the reviewer name.
+			const fetchReviewResourceDetails = await reviewResourcesQueries.findAll(
+				{
+					reviewer_id: user_id,
+				},
+				['resource_id', 'organization_id']
+			)
+			uniqueOrganizationIds = utils.getUniqueElements(
+				fetchReviewResourceDetails.map((item) => item.organization_id)
+			)
+
+			const findAndCount = await reviewsQueries.countDistinct(
+				{
+					organization_id: {
+						[Op.in]: uniqueOrganizationIds,
+					},
+					reviewer_id: user_id,
+					status: { [Op.in]: [common.REVIEW_STATUS_INPROGRESS, common.REVIEW_STATUS_CHANGES_UPDATED] },
+				},
+				['resource_id']
+			)
+
+			const in_progress_count = findAndCount.count
+			inProgressResources = utils.getUniqueElements(findAndCount.resource_ids)
+
+			if (common.STATUS in queryParams && queryParams[common.STATUS] === common.REVIEW_STATUS_INPROGRESS) {
+				finalResourceIds = inProgressResources
+			} else {
+				// fetch the resources types of an organization
+				let { sequential: resourceTypesInSequentialReview, parallel: resourceTypesInParallelReview } =
+					await this.fetchResourceReviewTypes(organization_id)
+
+				if (common.TYPE in queryParams && queryParams[common.TYPE]) {
+					resourceTypesInSequentialReview = queryParams[common.TYPE].split(',')
+					resourceTypesInParallelReview = queryParams[common.TYPE].split(',')
+				}
+
+				if (resourceTypesInSequentialReview.length > 0) {
+					// fetch all sequential resource ids from org
+					const sequentialResourcesIds = await this.findSequentialResources(
+						organization_id,
+						roles,
+						resourceTypesInSequentialReview
+					)
+					finalResourceIds = [...finalResourceIds, ...sequentialResourcesIds]
+
+					// fetch all resource ids assigned to another reviewer or reviewing by another reviewer
+					resourceIdsToBeRemoved = await this.findResourcesPickedUpByAnotherReviewer(
+						user_id,
+						finalResourceIds
+					)
+				}
+
+				if (resourceTypesInParallelReview.length > 0) {
+					// fetch all parallel resource ids from org
+					const parallelResourcesIds = await this.findParallelResources(
+						organization_id,
+						resourceTypesInParallelReview
+					)
+					finalResourceIds = [...finalResourceIds, ...parallelResourcesIds]
+				}
+
+				const assignedToMe = await this.findResourcesAssignedToReviewer(uniqueOrganizationIds, user_id)
+				finalResourceIds = [...finalResourceIds, ...assignedToMe]
+
+				finalResourceIds = _.difference(
+					utils.getUniqueElements(finalResourceIds),
+					utils.getUniqueElements(resourceIdsToBeRemoved)
+				)
+			}
+
+			const resourceFilter = {
+				id: { [Op.in]: finalResourceIds },
```

Comment:
@adithyadinesh0412 if finalResourceIds is greater than 0 then only do this, otherwise return empty